### PR TITLE
[FIX] web: keep server group order in pivot

### PR DESCRIPTION
**Pull request by Polymorphe57**

### CODE REVIEW

File: addons/web/static/src/js/views/pivot/pivot_model.js [hunk 1]
```diff
@@ -1,6 +1,8 @@
 odoo.define('web.PivotModel', function (require) {
 "use strict";
 
+/** @todo review doc below: directSubTrees becomes a Map*/
```
Comment (aab-odoo) [inline addons/web/static/src/js/views/pivot/pivot_model.js:4]: todo

File: addons/web/static/src/js/views/pivot/pivot_model.js [hunk 2]
```diff
@@ -335,7 +337,7 @@ var PivotModel = AbstractModel.extend({
             tree = this.colGroupTree;
             group = this._findGroup(this.colGroupTree, groupId[1]);
         }
-        group.directSubTrees = {};
+        group.directSubTrees.clear();
```
Comment (aab-odoo) [inline addons/web/static/src/js/views/pivot/pivot_model.js:340]: are you sure 'directSubTrees' always exists here?

Comment (Polymorphe57) [reply addons/web/static/src/js/views/pivot/pivot_model.js:340]: If the group exists (which should be the case), it is by construction in colGroupTree or rowGroupTree and at each level,
directSubTrees is a Map instance. To summarize, if group is defined, yes. Else we would have an error 'directSubTrees of undefined'. Do you expect this to happen?

Comment (Polymorphe57) [reply addons/web/static/src/js/views/pivot/pivot_model.js:340]: A concurency issue?

Comment (aab-odoo) [reply addons/web/static/src/js/views/pivot/pivot_model.js:340]: No, it's fine then.

Comment (Polymorphe57) [reply addons/web/static/src/js/views/pivot/pivot_model.js:340]: Can you robodoo r+ then or is there something else? Thank you

**Comment by aab-odoo** [comment]
robodoo r+

**Comment by aab-odoo** [comment]
robodoo r-

**Comment by aab-odoo** [comment]
robodoo r+

**Pull request closed without merging**